src/docfx/build/xref/XrefMap.cs

```diff
@@ -354,21 +359,102 @@ private static (List<Error> errors, InternalXrefSpec spec) LoadSchemaDocument(Co
+
+            foreach (var (uid, value) in uidToJsonPath)
             {
-                Uid = uid,
-                Href = file.CanonicalUrlWithoutLocale,
-                ReferencedFile = file,
-            };
+                extensionDataByUid.Add(uid, (string.IsNullOrEmpty(value), new Dictionary<string, Lazy<JValue>>()));
+            }
+
+            foreach (var (jsonPath, xrefProperty) in extensionData)
+            {
+                var (uid, resolvedJsonPath) = MatchExtensionDataToUid(jsonPath);
+                if (extensionDataByUid.ContainsKey(uid))
+                {
+                    var (_, properties) = extensionDataByUid[uid];
+                    properties.Add(resolvedJsonPath, xrefProperty);
+                }
+                else
+                {
+                    extensionDataByUid.Add(uid, (string.IsNullOrEmpty(uidToJsonPath[uid]), new Dictionary<string, Lazy<JValue>> { { resolvedJsonPath, xrefProperty } }));
+                }
+            }
+
+            var specs = extensionDataByUid.Select(item =>
+            {
+                var (isRoot, properties) = item.Value;
+                var xref = new InternalXrefSpec
+                {
+                    Uid = item.Key,
+                    Href = isRoot ? file.CanonicalUrlWithoutLocale : $"{file.CanonicalUrlWithoutLocale}#{GetBookmarkFromUid(item.Key)}",
+                    ReferencedFile = file,
+                };
+                xref.ExtensionData.AddRange(properties);
+                return xref;
+            }).ToList();
 
-            xref.ExtensionData.AddRange(extensionData);
-            return (errors, xref);
+            return (errors, specs);
+
+            string GetBookmarkFromUid(string uid)
+                => Regex.Replace(uid, @"\W", "_");
+
+            (string uid, string jsonPath) MatchExtensionDataToUid(string jsonPath)
+            {
+                string subString;
+                var index = jsonPath.LastIndexOf('.');
+                if (index == -1)
+                {
+                    subString = string.Empty;
+                }
+                else
+                {
+                    subString = jsonPath.Substring(0, index);
+                }
+
+                return jsonPathToUid.ContainsKey(subString) ? (jsonPathToUid[subString], jsonPath.Substring(index + 1)) : MatchExtensionDataToUid(subString);
+            }
+        }
+
+        private static void GetUids(Context context, string filePath, JObject token, Dictionary<string, string> uidToJsonPath, Dictionary<string, string> jsonPathToUid)
+        {
+            if (token is null)
+                return;
+
+            if (token.TryGetValue("uid", out var value) && value is JValue v && v.Value is string str)
+            {
+                if (!uidToJsonPath.TryAdd(str, token.Path))
+                {
+                    context.Report.Write(filePath, Errors.UidConflict(str));
```

Comment:
>context [](start = 20, length = 7)

does it only woking for inside conflicts?

Reply:
Here we only want to detect duplicated uids defined in one file.
The case missing is, detect duplicated uid defined already externally, defined again in the build repo

Reply:
no need comparing with external uids, but make sure the uid is unique inside this docset. add test for that

---
In reply to: [250914849](https://github.com/dotnet/docfx/pull/3934#discussion_r250914849) [](ancestors = 250914849)

Reply:
test added